lib/secp256k1/privatekey.ex

```diff
@@ -218,4 +218,20 @@ defmodule Bitcoinex.Secp256k1.PrivateKey do
         end
     end
   end
+
+  @doc """
+  sign_message returns an ECDSA signature using the privkey and msg
+  where privkey is a PrivateKey object and msg is a binary message to be hashed. 
+  The message is hashed using hash256 (double SHA256) and the nonce is derived 
+  using RFC6979. 
+  """
+  @spec sign(t(), binary) :: Signature.t()
+  def sign_message(privkey, msg) do
```

Comment:
If `sign_message` is expected to accept an arbitrary message. The message should be formatted with `Bitcoin Signed Message: <msg>` so that the user does not mistakenly sign an actual transaction.

Reply:
updated to match bitcoin core.